cmd/zdb/zdb.c

```diff
@@ -4619,20 +4618,25 @@ dump_l2arc_log_blocks(int fd, const l2arc_dev_hdr_phys_t *l2dhdr,
 		switch (L2BLK_GET_COMPRESS((&lbps[0])->lbp_prop)) {
 		case ZIO_COMPRESS_OFF:
 			break;
-		default:
-			abd = abd_alloc_for_io(asize, B_TRUE);
+		default: {
+			abd_t *abd = abd_alloc_for_io(asize, B_TRUE);
```

Comment:
We could use `abd_alloc_linear()` here, since it is transient and never actually goes through I/O.

Reply:
Fair. This one I'd left because I wasn't there to modify existing code, but why not if you say so!